test/unit/services/awsAdapter_spec.js

```diff
@@ -564,6 +565,157 @@ describe("The AwsAdapter class ", function () {
+		});
+
+		describe("faces an error while polling and tries to update an instance which has already been modified",
+		function () {
+			var SshPollingStub;
+			var getPublicIPAddressStub;
+			var instanceProp;
+			var result;
+
+			before(function () {
+				SshPollingStub = Sinon.stub(sshAdapter,"SshPolling", function () {
+					return Bluebird.reject("Simulated Failure.");
+				});
+
+				getPublicIPAddressStub = Sinon.stub(awsAdapter,"getPublicIPAddress", function () {
+					return Bluebird.resolve(VALID_IP_ADDRESS);
+				});
+
+				return instances.createInstance("ami-d05e75b8","t2.micro")
+				.then(function (data) {
+					var updatedInstance = new Instance({
+						id    : data.id,
+						type  : data.type,
+						ami   : data.ami,
+						state : "terminating",
+						uri   : data.uri
+					});
+					instances.updateInstance(updatedInstance);
+					instanceProp  = JSON.parse(JSON.stringify(data));
+					instanceProp.instanceID = VALID_EC2_INSTANCE;
+					return awsAdapter.beginPolling(instanceProp);
+				})
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			after(function () {
+				SshPollingStub.restore();
+				getPublicIPAddressStub.restore();
+			});
+
+			it("does not update the current instance status", function () {
+				expect(result.id).to.equal(instanceProp.id);
+				expect(result.type).to.equal(instanceProp.type);
+				expect(result.ami).to.equal(instanceProp.ami);
+				expect(result.state).to.equal("terminating");
+				expect(result.uri).to.be.null;
+			});
+		});
+
+		describe("faces an error while polling and getInstance does not show state has been updated",
+		function () {
+			var SshPollingStub;
+			var getPublicIPAddressStub;
+			var getInstanceStub;
+			var instanceProp;
+			var result;
+
+			before(function () {
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceProp = instance;
+					SshPollingStub = Sinon.stub(sshAdapter,"SshPolling", function () {
+						return Bluebird.reject("Simulated Failure.");
+					});
+
+					getPublicIPAddressStub = Sinon.stub(awsAdapter,"getPublicIPAddress", function () {
+						return Bluebird.resolve(VALID_IP_ADDRESS);
+					});
+
+					getInstanceStub = Sinon.stub(instances, "getInstance", function () {
+						return instances.updateInstance(instance)
+						.then(function (response) {
+							return Bluebird.resolve({
+								id       : response.id,
+								ami      : response.ami,
+								type     : response.type,
+								state    : response.state,
+								uri      : response.uri,
+								revision : response.revision - 1
+							});
+						});
+					});
+				})
+				.then(function () {
+					return awsAdapter.beginPolling(instanceProp);
+				})
+				.catch (function (err) {
+					result = err;
+				});
+			});
+
+			after(function () {
+				SshPollingStub.restore();
+				getPublicIPAddressStub.restore();
+				getInstanceStub.restore();
+			});
+
+			it("fails to update as it has wrong revision number", function () {
```

Comment:
This is a test case where a getInstance is performed, an update instance is done (by a PUT request) and then an updateInstance is done based on the details received from the getInstance.